src/functions/deduplicateLeads.js

```diff
@@ -0,0 +1,24 @@
+const { writeLog } = require("./logDuplicates");
+
+const deduplicateLeads = (leadsArr) => {
+  return leadsArr
+    .slice()
+    .reverse()
+    .filter((lead, index, arr) => {
+      const duplicateID = arr.findIndex((t) => t._id === lead._id) !== index;
```

Comment:
I changed your function in a few ways, and wanted to leave you with a couple things to consider as well:

1. I flipped the conditions of your `duplicateID` and `duplicateEmail` checks because it's much more intuitive to read the logic as "if a duplicate ID or duplicate email is found, print it out"
2. Never use `array` as an argument name. It's too close to the language's reserved keyword `Array`, and while it's not the exactly the same, you're playing with fire. Why not be more clear and use `leads` or `leadsArr` since that's what we're dealing with?
3. I explicitly returned `false` from the cases in which you're printing the duplicates because it's easy to miss that those items won't be included in the array. While it's not necessary in order to filter out the dups, I think the explicit return signals more intent.
4. You don't need to access the current value via `arr[index]` -- you have the current value available as the first argument of your `forEach` callback function already
5. Have you learned about the concept of "pure functions"? I'm not a huge fan of performing the printouts in the `.filter` call -- those are what are known as "side effects": rather the function taking an input and strictly returning an input (having one simple job), it now is also responsible for printing certain things out on the way as well. It's a prime spot for bugs to be introduced because different logic is getting jumbled together.

If I were writing this from scratch, I'd probably have initialized a few different arrays (`duplicateIDs`, duplicateEmails`, and `deduplicatedLeads`), used a traditional for loop to go through them, pushed to the appropriate array  in the branching logic area, and then run two separate `forEach` loops on the duplicate arrays to print out all contents before ultimately deduplicated leads array at the bottom of the function.
However, this ultimately works and I think it's still pretty readable, so I think what you have is still solid overall, but it's something to keep in mind for the future.